client/src/functions/calcTotalPrice.js

```diff
@@ -0,0 +1,23 @@
+import PropTypes from "prop-types";
+
+function calcTotalPrice({ listProductIDs, supermarket }) {
+  const totalPrice = listProductIDs.reduce((totalPrice, val) => {
+    const supermarketProduct = supermarket.products.find(
+      (product) => product.productId === val
+    );
+    if (supermarketProduct) {
+      return totalPrice + supermarketProduct.price;
+    } else {
+      return totalPrice;
+    }
```

Comment:
As an alternative (up to you)
```suggestion
    return supermarketProduct ? totalPrice + supermarketProduct.price : totalPrice;
```